src/test/java/com/expediagroup/hiveberg/TestIcebergInputFormat.java

```diff
@@ -53,35 +60,63 @@ public void before() throws IOException {
         optional(2, "salary", Types.LongType.get()));
     PartitionSpec spec = PartitionSpec.unpartitioned();
     HadoopTables tables = new HadoopTables();
-    Table table = tables.create(schema, spec, tableLocation.getAbsolutePath());
+    table = tables.create(schema, spec, tableLocation.getAbsolutePath());
 
     DataFile fileA = DataFiles
         .builder(spec)
         .withPath("src/test/resources/test-table/data/00000-1-c7557bc3-ae0d-46fb-804e-e9806abf81c7-00000.parquet")
-        .withFileSizeInBytes(0)
+        .withFileSizeInBytes(1024)
         .withRecordCount(3) // needs at least one record or else metrics will filter it out
         .build();
+
     table.newAppend().appendFile(fileA).commit();
   }
 
   @Test
-  public void testGetSplits() {
+  public void testInputFormat() {
     shell.execute("CREATE DATABASE source_db");
-    shell
-        .execute(new StringBuilder()
+    shell.execute(new StringBuilder()
             .append("CREATE TABLE source_db.table_a ")
-            // .append("(name STRING, salary INT) ")
-            .append("ROW FORMAT SERDE 'org.apache.hadoop.hive.serde2.avro.AvroSerDe' ")
+            .append("ROW FORMAT SERDE 'com.expediagroup.hiveberg.IcebergSerDe' ")
             .append("STORED AS ")
             .append("INPUTFORMAT 'com.expediagroup.hiveberg.IcebergInputFormat' ")
             .append("OUTPUTFORMAT 'org.apache.hadoop.hive.ql.io.HiveIgnoreKeyTextOutputFormat' ")
             .append("LOCATION '")
             .append(tableLocation.getAbsolutePath())
-            .append(
-                "' TBLPROPERTIES ('avro.schema.literal' = '{\"namespace\":\"source_db\",\"type\":\"record\",\"name\":\"table_a\",\"fields\":[{\"name\":\"name\",\"type\":[\"null\",\"string\"],\"default\":null,\"field_id\":1},{\"name\":\"salary\",\"type\":[\"null\",\"long\"],\"default\":null,\"field_id\":2}]}')")
+            .append("'")
             .toString());
 
-    List<Object[]> result = shell.executeStatement("SELECT * FROM source_db.table_a WHERE salary=3000");
+    List<Object[]> result = shell.executeStatement("SELECT * FROM source_db.table_a");
```

Comment:
Do we want to assert the results or are we just checking that this doesn't throw any exceptions?

Reply:
Adding assertions for this and the SchemaToTypeInfo util class